The goal of this EDA is surface level data analysis, with a focus on churn distribution, and price sensitivity.

- is there a correlation between price/power usage, and churn?
- any noticeable patterns among inactive accounts?

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

client_data = pd.read_csv('client_data.csv')
price_data = pd.read_csv('price_data.csv')

In [ ]:
#print(client_data.info())
#print(price_data.info())
#print(client_data.describe())
#print(price_data.describe())
#print(client_data['id'].nunique())
#print(client_data['id'].count())

In [23]:
print(client_data['churn'].value_counts())
print(f'\nPercentage of active customers: {client_data['churn'].value_counts(normalize=True)[0] * 100:.2f}%')
print(f'Percentage of churned customers: {client_data['churn'].value_counts(normalize=True)[1] * 100:.2f}%')

#plt.figure(figsize=(8, 6))
#sns.countplot(x='churn', data=client_data, hue='churn')
#plt.title('Churn Distribution')

churn
0    13187
1     1419
Name: count, dtype: int64

Percentage of active customers: 90.28%
Percentage of churned customers: 9.72%


- There are no duplicate customer ID's, **14606 unique values**
- **13187 active** (90.28%) (churn = 0), **1419 inactive** (9.72%) (churn = 1)